### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by toote**
* Splits Readme into 2 files: one with options, another with examples
* Rewording of all options
* review of plugin schema to ensure all options are properly validated
* update of the pipeline to validate examples in the new file
* bump version to 5.0.0 in preparation of upcoming major release
* adds a new document with what I could think may help users migrate to this new major version

### CODE REVIEW

File: plugin.yml [hunk 1]
```diff
@@ -14,6 +14,7 @@ configuration:
     push:
       type: [ string, array ]
       minimum: 1
+
```
Comment (tomowatt) [inline plugin.yml:17]: Checking this additional line is expected to break up the plugin "commands" from the additional arguements?

**Review by tomowatt — APPROVED** [review]
Looks good to me, only a comment but not a blocker

File: plugin.yml (hunk 1 above)
Comment (toote) [reply plugin.yml:17]: I added it on purpose to separate the _"main commands"_ from the rest of the options that are sorted alphabetically

File: README.md
[the remark this answers is not in the document]
Comment (toote) [reply README.md:1]: good catch, I went through all options and made sure that:
* all options are documented
* all documented options are in the plugin's schema
* all options that are not just a string has valid types in the documentation and schema

**Pull request merged**